mne/decoding/search_light.py

```diff
@@ -0,0 +1,491 @@
+# Author: Jean-Remi King <jeanremi.king@gmail.com>
+#
+# License: BSD (3-clause)
+
+import numpy as np
+
+from .mixin import TransformerMixin
+from .base import BaseEstimator  # XXX why are these sklearn object copied in weird places. We should just have an externals.sklearn? # noqa
```

Comment:
it's because we did not want to have sklearn as hard dependencies and adding full sklearn code in externals make packaging more complecated. Debian for example would require us to strip this code out for packaging.